### ISSUE TITLE
drpcli machines processjobs - reestablish connection on IP change

### ISSUE THREAD

**Issue by ytjohn** [digitalrebar/provision#1018]
If you create a reservation for a sledgehammer-bootenv machine or otherwise cause the IP address to change, `drpcli machines processjobs` will drop its websocket connection to the DRP server. It will no longer be able to process jobs. Ideally drpcli would recognize that the websocket has dropped connection and re-establish connection. 

**To reproduce:**

1. Boot a machine into sledgehammer discovery
2. Create a reservation for this machine's mac address with a different IP.
3. After the TTL expires, machine will pick up the new IP. You can see this in leases.
4. Attempt to run a new workflow. That uses the sledgehammer bootenv. The first task will no proceed.

**Workarounds**

Option 1: ssh into the machine and run `drpcli machines processjobs $RS_UUID`. This instance will immediately start processing the pending tasks. This validates connectivity back to DRP is available and the workflow would have ran if the existing drpcli process had reconnected.

Option 2:. Reboot the machine. It will reboot back into sledgehammer with the new and restart `drpcli machines processjobs {{.Machine.UUID }}`

**Comment by ytjohn** [comment]
Additional workaround:

Create a stage, task, template to lay down dhcp exit hooks. If 

Have a task drop this into `/etc/dhcp/dhclient-exit-hooks.d/drpcli-runner.sh` and another task make it executable. Roll those into a stage that gets ran (I add it to my discovery workflow) before any other workflow would create the dhcp reservation. 

```bash
#!/usr/bin/env bash
# If the server acquires a new IP address, existing drpcli
# processjobs will stop running. This hook should tart a new instance.

case "$reason" in
BOUND|REBOOT)
echo "dhcpdrp $reason, starting new drpcli machines processjobs" | logger

# killall not found, brew your own
for pid in `ps aux | egrep '[d]rpcli machines processjobs' | awk '{print $2}'`; do echo "killing $pid"; kill -9 $pid; done
# run control.sh which starts a drpcli machines processjobs instance
nohup /tmp/control.sh | logger &

;;

RENEW|REBIND)
echo "dhcpdrp $reason, doing nothing" | logger

esac

```

**Comment by sygibson** [comment]
Currently evaluating solutions to resolve this issue.
